### ISSUE TITLE
@types/node pollution

### ISSUE THREAD

**Issue by kingyue737** [kettanaito/msw#1361]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Browsers

Chromium (Chrome, Brave, etc.)

### Reproduction repository

Unnecessary

### Reproduction steps

Just import msw in a typescript project

### Current behavior

Import `msw` will also import `@types/node` which pollutes types.

For example, after import `msw`, type of `setTimeout` is no longer correct:

![image](https://user-images.githubusercontent.com/40021217/183258945-0f28d77b-c90b-4730-b8d6-13758bd4fd56.png)

This issue results from type declaration of [`strict-event-emitter`](https://unpkg.com/browse/strict-event-emitter@0.2.4/lib/StrictEventEmitter.d.ts):

![image](https://user-images.githubusercontent.com/40021217/183259195-9a1c86df-c351-4118-88d3-931b380d9e41.png)

Removing the first line can fix the bug.

### Expected behavior

Not import types of node when using the library in a browser

**Comment by kettanaito** [comment]
Hey, @kingyue737. Thanks for reporting this.

The real pickle is why `strict-events-emitter` even ends up in a browser runtime. There's no `EventEmitter` in a browser, so I suspect the code shouldn't end up there. 

The type declarations for `strict-event-emitter` itself are correct: it's a library for type-safe `events.EventEmitter` for Node.js. What I think is the issue here is that SEE ends up in the `msw` bundle no matter what. Perhaps a code splitting problem? 

As always, any help on this is welcome.

**Comment by kingyue737** [comment]
Sorry I'm new to msw. If there is no `EventEmitter` in browser, why worker returned by `setupWorker()` has `events` property?

![image](https://user-images.githubusercontent.com/40021217/183566754-193be232-bc51-4463-8d3d-31df6d05d978.png)

**Comment by kettanaito** [comment]
I don't think this is strictly an MSW-specific issue. Looks like it's related to how we annotate emitters. And yet somehow it works. Interesting. I'd expect we use `EventTarget` in browsers. Or, even better, use `StrictEventEmitter` as an interface which each implementation can implement.

```ts
// node.js
export class StrictEmitter extends EventEmitter implements StrictEventEmitter<EventsMap> {}
```

```ts
// browser.js
export class StrictEmitter extends EventTarget implements StrictEventEmitter<EventsMap> {}
```

Does this make sense to you?

**Comment by judicaelandria** [comment]
Actually, the events that come from setupWorker are already using the type from EventTarget, 
I mean it is the expected behavior, right?

**Comment by jkieboom** [comment]
I am also running into this issue and I'm blocked on this for integrating msw. `StrictEventEmitter` ends up in the typings and as the issue describes it explicitly references node typings. I am a bit confused by the discussion on this issue. `setupWorker` does `new StrictEventEmitter` .

**Comment by jkieboom** [comment]
I had a quick look, and I think this is caused by the fact that `strict-event-emitter` does not install `@types/events` so it's picking up the typings from the nodejs native events module, and so inserts the `types/node` reference.

**Comment by jkieboom** [comment]
https://github.com/open-draft/strict-event-emitter/pull/5

**Comment by kingyue737** [comment]
@jkieboom  It does not appear to be fixed. `/// <reference types="node" />` is still there 

https://www.unpkg.com/browse/strict-event-emitter@0.2.7/lib/StrictEventEmitter.d.ts

**Comment by jkieboom** [comment]
Hmm, not sure what I did when I tested it locally, it's indeed still there also for me. We also need to set `types: []` in the `tsconfig`. I'll prepare another PR.
